esp/src/src-react/components/WorkunitDetails.tsx

```diff
@@ -104,12 +123,88 @@ export const WorkunitDetails: React.FunctionComponent<WorkunitDetailsProps> = ({
+            key: "delete", text: nlsHPCC.Delete, iconProps: { iconName: "Delete" }, disabled: !canDelete,
             onClick: () => {
-                navigator?.clipboard?.writeText(wuid);
+                if (confirm(nlsHPCC.YouAreAboutToDeleteThisWorkunit)) {
+                    workunit?.delete();
+                    pushUrl("/workunits/");
+                }
             }
         },
+        {
+            key: "restore", text: nlsHPCC.Restore, disabled: !workunit?.Archived,
+            onClick: () => workunit?.restore()
+        },
         { key: "divider_2", itemType: ContextualMenuItemType.Divider, onRender: () => <ShortVerticalDivider /> },
+        {
+            key: "reschedule", text: nlsHPCC.Reschedule, disabled: !canReschedule,
+            onClick: () => workunit?.reschedule()
+        },
+        {
+            key: "deschedule", text: nlsHPCC.Deschedule, disabled: !canDeschedule,
+            onClick: () => workunit?.deschedule()
+        },
+        { key: "divider_3", itemType: ContextualMenuItemType.Divider, onRender: () => <ShortVerticalDivider /> },
+        {
+            key: "setToFailed", text: nlsHPCC.SetToFailed, disabled: workunit?.Archived || workunit?.isComplete() || workunit?.isDeleted(),
+            onClick: () => workunit?.setToFailed()
+        },
+        {
+            key: "abort", text: nlsHPCC.Abort, disabled: workunit?.Archived || workunit?.isComplete() || workunit?.isDeleted(),
+            onClick: () => workunit?.abort()
+        },
+        { key: "divider_4", itemType: ContextualMenuItemType.Divider, onRender: () => <ShortVerticalDivider /> },
+        {
+            key: "recover", text: nlsHPCC.Recover, disabled: workunit?.Archived || !workunit?.isComplete() || workunit?.isDeleted(),
+            onClick: () => workunit?.resubmit()
+        },
+        {
+            key: "resubmit", text: nlsHPCC.Resubmit, disabled: workunit?.Archived || !workunit?.isComplete() || workunit?.isDeleted(),
+            onClick: () => workunit?.resubmit()
+        },
+        {
+            key: "clone", text: nlsHPCC.Clone, disabled: workunit?.Archived || !workunit?.isComplete() || workunit?.isDeleted(),
+            onClick: () => {
+                // there seems to be an issue with clone() returning a 404 when
+                // attempting to refresh / query the status of the new workunit
+                // workunit?.clone().then(wu => {
+                //     if (wu && wu.Wuid) {
+                //         pushUrl(`/workunits/${wu?.Wuid}`);
+                //     }
+                // });
+                const formData = new FormData();
+                formData.append("Wuids", wuid);
+                formData.append("ResetWorkflow", "0");
+                formData.append("CloneWorkunit", "1");
+                fetch("/WsWorkunits/WUResubmit.json", {
+                    method: "POST",
+                    body: formData
+                })
```

Comment:
If there isn't a resubmit method on workunit, you could add it into `@hpcc-js/comms`?

Reply:
There is one already, but it doesn't accept a param for CloneWorkunit. This was just a workaround for the issue with clone()

Reply:
@jeclrsg - This fix should be released now.